x-pack/legacy/plugins/ml/server/models/data_frame/transform_audit_messages.ts

```diff
@@ -4,49 +4,59 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-
 import { ML_DF_NOTIFICATION_INDEX_PATTERN } from '../../../common/constants/index_patterns';
+import { callWithRequestType } from '../../../common/types/kibana';
+import { TransformMessage } from '../../../common/types/messages';
 
 const SIZE = 1000;
 
-export function transformAuditMessagesProvider(callWithRequest) {
+interface Message {
+  _index: string;
+  _type: string;
+  _id: string;
+  _score: null | number;
+  _source: TransformMessage;
+  sort?: any;
+}
 
+export function transformAuditMessagesProvider(callWithRequest: callWithRequestType) {
   // search for audit messages,
   // transformId is optional. without it, all jobs will be listed.
-  async function getTransformAuditMessages(transformId) {
+  async function getTransformAuditMessages(transformId: string) {
     const query = {
       bool: {
         filter: [
           {
             bool: {
               must_not: {
                 term: {
-                  level: 'activity'
-                }
-              }
-            }
+                  level: 'activity',
+                },
+              },
+            },
           },
-        ]
-      }
+        ],
+      },
     };
 
     // if no transformId specified, load all of the messages
     if (transformId !== undefined) {
       query.bool.filter.push({
         bool: {
+          // @ts-ignore
```

Comment:
nit: suggest to not use `@ts-ignore` here but instead cast `query` to a more general type so it accepts this push.

Reply:
Updated in 35ad48d475c32fd3c7800afabfa2c4bb5985ed99